src/main/java/com/iexec/blockchain/broker/BrokerService.java

```diff
@@ -115,41 +112,81 @@ private Optional<String> fireMatchOrders(BrokerOrder brokerOrder) {
         return Optional.empty();
     }
 
+    void checkRequestOrder(RequestOrder requestOrder, boolean withDataset) {
```

Comment:
Can't we retrieve `withDataset` from `requestOrder`? (requestOrder.dataset==null)

Reply:
Fixed in 11cf4bc